c++/src/vector.cc

```diff
@@ -44,18 +45,26 @@ class Vector {
     return this->dataStore_[length_--];
   }
 
-  unsigned int length() const{
-    return this->length_;
-  }
+  void reserve(unsigned int reserveIndices) {
+    while (this->capacity_ < reserveIndices) {
+      this->capacity_ = this->capacity_ * this->growFactor;
+    }
```

Comment:
If you do this you'll probably want to add tests for testing when the requested reserve capacity is a power of 2, and when it's not, perhaps with those example inputs above: 128 => 128 and 129 => 256.